docs/DesignDocs/ExtendedEnclaveInitializationData.md

```diff
@@ -0,0 +1,168 @@
+Extended Enclave Initialization Data
+=====
+
+We present an extension to attestation in OpenEnclave that enables the attestation of enclave initialization parameters.
+Our extension uses TEE evidence (e.g. MRENCLAVE in SGX) as a platform configuration register to measure additional data besides the signed enclave image, such as instance-specific enclave configuration and the settings in the current enclave signing configuration.
+
+We introduce a new OE evidence type that holds extended enclave initialization data (EEID).
+Recipients of such evidence must be able to recover and verify the original enclave identity information (including the original evidence, like MRENCLAVE and MRSIGNER in the case of SGX), based on the knowledge of the EEID, which is transmitted alongside the evidence as a new type of endorsement.
+
+
+Motivations
+----------
+
+In today's OE evidence, applications can attest their configuration as part of the `report_data` argument of `oe_get_report`.
+This mechanism works robustly when the application code is entirely and statically known to the verifier of the evidence.
+
+Things get more complicated if an enclave executes user code that is not part of the signed enclave image. For instance, consider an enclave containing a JavaScript interpreter that executes user scripts that originate from the untrusted host in shared enclave memory, and which has access to the `oe_get_report` API. It is impossible to know which script has been executed by such an enclave based on traditional evidence, even if the hash of the script is included in the `report_data`, because a malicious script can obtain valid evidence via `oe_get_report`, while pretending to be an honest script. Similarly, if an enclave loads and executes arbitrary assembly code from the host, this assembly code can use an in-enclave CPU instruction, for instance the `EREPORT` instruction on SGX, to create valid evidence - hence, it is impossible to use traditional attestation to determine which assembly code was executed by the enclave.
+
```

Comment:
In theory a script could modify the `oe_eeid_t `, but attestation would fail because the corresponding EPC pages, and thus the enclave hash, are modified

Reply:
Yes, in the case of SGX that's the MRENCLAVE, I assumed "enclave hash" was the platform-independent term for this. Is there a better term?